src/trace/context.ts

```diff
@@ -238,6 +237,136 @@ export function readTraceFromSQSEvent(event: SQSEvent): TraceContext | undefined
   return;
 }
 
+export function readTraceFromSNSSQSEvent(event: SQSEvent): TraceContext | undefined {
+  if (event?.Records[0]?.body) {
```

Comment:
It's very unlikely that `Records` here will be `null`, but the proper syntax is the weird looking `event?.Records?.[0]?.body`.

```
const testEvent1 = {Records: [{body: "Hello"}]}
console.log("testEvent1?.Records?.[0]?.body:", testEvent1?.Records?.[0]?.body)
console.log("testEvent1?.Records[0]?.body:", testEvent1?.Records[0]?.body)

const testEvent2 = {Records: []}
console.log("testEvent2?.Records?.[0]?.body:", testEvent2?.Records?.[0]?.body)
console.log("testEvent2?.Records[0]?.body:", testEvent2?.Records[0]?.body)

const testEvent3 = {Records: null}
console.log("testEvent3?.Records?.[0]?.body:", testEvent3?.Records?.[0]?.body)
console.log("testEvent3?.Records[0]?.body:", testEvent3?.Records[0]?.body)
```

```
/usr/local/lib/node_modules/ts-node-fm/src/index.ts:226
testEvent1?.Records?.[0]?.body: Hello
testEvent1?.Records[0]?.body: Hello
testEvent2?.Records?.[0]?.body: undefined
testEvent2?.Records[0]?.body: undefined
testEvent3?.Records?.[0]?.body: undefined
TypeError: Cannot read property '0' of null
    at index.ts:10:126
    at Script.runInThisContext (vm.js:122:20)
    at startRepl (/usr/local/lib/node_modules/ts-node-fm/src/bin.ts:157:12)
    at Object.<anonymous> (/usr/local/lib/node_modules/ts-node-fm/src/bin.ts:66:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
```

Reply:
Yeah... But it's ugly. I'll use it :)